src/AutoMapper/PropertyMap.cs

```diff
@@ -112,6 +115,39 @@ internal void Seal()
                 (inner, res) => ctxt => res.Resolve(inner(ctxt), ctxt));
 
             SourceType = resolvers.OfType<IMemberResolver>().LastOrDefault()?.MemberType;
+
+            if (resolvers.All(r => r is IMemberResolver)

```

Comment:
Maybe IExpressionResolver here? IMemberResolver has the type and I needed the type to remove the ResolutionResult so now the DelegateBasedResolver implements IMemberResolver :)